File: src/main/java/org/jetbrains/fortran/FortranFixedFormFileTypeFactory.java
package org.jetbrains.fortran;

import com.intellij.openapi.fileTypes.FileTypeConsumer;
import com.intellij.openapi.fileTypes.FileTypeFactory;
import org.jetbrains.annotations.NotNull;

public class FortranFixedFormFileTypeFactory extends FileTypeFactory {
    @Override
    public void createFileTypes(@NotNull FileTypeConsumer fileTypeConsumer) {
        for (int i = 0; i < FortranFixedFormFileType.DEFAULT_EXTENSIONS.length; i++)
        {
            fileTypeConsumer.consume(FortranFixedFormFileType.INSTANCE, FortranFixedFormFileType.DEFAULT_EXTENSIONS[i]);
        }
    }
}

File: src/main/java/org/jetbrains/fortran/FortranFixedFormFileTypeFactory.kt
package org.jetbrains.fortran

import com.intellij.openapi.fileTypes.FileTypeConsumer
import com.intellij.openapi.fileTypes.FileTypeFactory

class FortranFixedFormFileTypeFactory : FileTypeFactory() {
    override fun createFileTypes(fileTypeConsumer: FileTypeConsumer) {
        for (i in FortranFixedFormFileType.DEFAULT_EXTENSIONS.indices) {
            fileTypeConsumer.consume(FortranFixedFormFileType, FortranFixedFormFileType.DEFAULT_EXTENSIONS[i])
        }
    }
}
